src/librustc/session/config.rs

```diff
@@ -2557,6 +2563,73 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
         remap_path_prefix,
         edition,
         json_artifact_notifications,
+        pretty,
+    }
+}
+
+fn parse_pretty(
```

Comment:
Aw :( -- Ideally we should move stuff out of librustc instead of the other direction... perhaps we can move most of this file out of librustc...

Reply:
I think it's fine to do so but it's also a pretty involved change (perhaps we want to create a `librustc_pretty` or something) -- I don't think it should be done in this PR.

Reply:
Yea; didn't mean in this PR. :)